app/components/edition/reorder_component.rb

```diff
@@ -0,0 +1,21 @@
+class Edition::ReorderComponent < ViewComponent::Base
+  def initialize(edition:, order:, redirect_path:)
+    @edition = edition
+    @order = order
+    @redirect_path = redirect_path
+  end
+
+private
+
+  attr_reader :edition, :order, :redirect_path
+
+  def move_path(index, item, direction)
+    order_dup = order.dup
+    new_index = direction == :up ? index - 1 : index + 1
+    order_dup.insert(
+      new_index,
+      order_dup.delete_at(order_dup.index(item)),
```

Comment:
Sorry, you'll have to bear with me as I get used to rails. This line struck me as though it might be complex to delete at an index by getting something by looking up its index. I thought there might maybe be something available in the api like so?:
```ruby
order_dup.delete(item)
```